pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -230,20 +232,22 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
                 "Segment file (as multipart/form-data) is required for SEGMENT upload mode",
                 Response.Status.BAD_REQUEST);
           }
-          if (!moveSegmentToFinalLocation && StringUtils.isEmpty(downloadURI)) {
+          if (!moveSegmentToFinalLocation && StringUtils.isEmpty(sourceDownloadURIStr)) {
```

Comment:
Not introduced in this PR, but consider changing `moveSegmentToFinalLocation` to `copySegmentToDeepStore`